src/presentation/fashion/CartDetail/CartLikePriceDetail/Promotions/Promotions.vue

```diff
@@ -0,0 +1,10 @@
+<style src="./Promotions.scss" lang="scss"></style>
+<script src="./Promotions.js"></script>
+
+<template>
+<span class="promo-title">Available promotions</span>
+<div class="code-container" v-for="code in availablePromotions" :key="code">
```

Comment:
Consider adding something like 
```html
<div class="code-container" v-for="code in availablePromotions" :key="code">
  <b>{{code.banner}}</b>
  <b class="code-gap"></b>
  <a href="#" class="add-promo" @click.prevent="() => handlePromotion(code)">ADD</a>
</div>
```

And styles 

```css
.code-container {
    display: flex;
    flex-wrap: wrap;
    flex-grow: 1;
}
.code-gap {
    flex-grow: 1;
}
```
To obtain similar view like this under Applied discounts.